tasks/insights-client.yml

```diff
@@ -94,6 +94,62 @@
           or "Registered" in __rhc_insights_status.stdout
       changed_when: true
 
+- name: Check display name in insights-client config
+  when:
+    - rhc_insights.display_name is defined
+    - not rhc_insights.display_name is none
+    - rhc_insights.display_name != __rhc_state_absent
+    - rhc_insights.display_name != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^display_name="
+    state: present
+    line: display_name={{ rhc_insights.display_name }}
+  check_mode: true
+  register: __insights_display_name_exists
+
+- name: Configure display name
+  when:
+    - rhc_insights.display_name is defined
+    - not rhc_insights.display_name is none
+    - rhc_insights.display_name != omit
+    - rhc_insights.display_name != __rhc_state_absent
+    - __insights_display_name_exists.changed
+  block:
+    - name: Update display name in insights-client config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^display_name="
+        insertafter: "#display_name="
+        line: display_name={{ rhc_insights.display_name }}
+    - name: Update display name in inventory
+      shell: >-
+        insights-client --display-name={{ rhc_insights.display_name }} & wait
+      when:
+        - >-
+          "This host is registered" in __rhc_insights_status.stdout
+          or "Registered" in __rhc_insights_status.stdout
+      changed_when: true
+
+- name: Remove display name
+  when:
+    - rhc_insights.display_name is defined
+    - rhc_insights.display_name == __rhc_state_absent
+  block:
+    - name: Remove display name from insights-client config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^display_name="
+        state: absent
+    - name: Remove display name in inventory
+      shell: insights-client --display-name= & wait
```

Comment:
Resetting the display name using this command doesn't seem to be working as expected in insights-client. It just starts another insights collection instead of reverting the display-name back to the system name which causes this to silently fail.

I'll have a look at the insights-client code that handles this, because it looks like it doesn't handle empty values being passed through like the `ansible-host` command.

Reply:
Oops didn't notice Pino already posted about this issue!